### PR TITLE
Remove deprecation of returning `channel.write*` & `writef` methods

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
This is the last of a few PRs related to the boolean-returning variants of these methods:
-write
-writeln
-writebits
-writeBytes
-writef in IO.FormattedIO
-writef in ChapelIO (top level)

- **[#20508](https://github.com/chapel-lang/chapel/pull/20508):** deprecated the returning methods using a config-param to control which methods are called (return-void or return-bool)
- We then decided that this deprecation is too noisy given that the return value is always true (it provides no information), and is very rarely used. In other words, removing the return value is more of a bug-fix than a deprecation.
- **[#20941](https://github.com/chapel-lang/chapel/pull/20941):** switched the default value of the `WritersReturnBool` flag to `false` as a temporary measure for reducing noise
    - Also added a case-specific error message to the compiler for rare cases when the return value is used
- **This PR:** removes the deprecation code s.t. we only have void-returning versions of the relevant methods
    - updates the case-specific error message to indicate how a user might achieve the desired behavior (i.e., catch an early EOF when writing to a channel).

Passing full paratest.

### CODE REVIEW

**Comment by vasslitvinov** [comment]
@jeremiah-corrado do you need to update start_test.py as part of this PR? Otherwise I do not see how a paratest would have succeeded.

**Comment by jeremiah-corrado** [comment]
Hi Vass, I removed my initial modification of `start_test` in #20941. Sorry, I should have made a note of that.

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -7703,19 +7579,11 @@ proc _channel.writef(fmtStr: ?t, const args ...?k): bool throws
    :arg args: 0 or more arguments to write
 
    :throws IllegalArgumentError: if an unsupported argument type is encountered.
-   :throws SystemError: if the arguments could not be written.
+   :throws SystemError: if the arguments could not be written.
```
Comment (vasslitvinov) [inline modules/standard/IO.chpl:7582]: Please remove the extra space at the end of the line.

**Review by vasslitvinov — APPROVED** [review]
Looks good. Please make the minor change in my comment.

Please run a gasnet paratest before or after merging, for better testing coverage.

**Comment by jeremiah-corrado** [comment]
Thanks for the review @vasslitvinov! I'll run another paratest with gasnet before merging.

**Pull request merged**